tecs/tecs.cpp

```diff
@@ -573,10 +573,38 @@ void TECS::_initialize_states(float pitch, float throttle_cruise, float baro_alt
 void TECS::_update_STE_rate_lim()
 {
 	// Calculate the specific total energy upper rate limits from the max throttle climb rate
-	_STE_rate_max = _max_climb_rate * CONSTANTS_ONE_G;
+	const float rate_max = _max_climb_rate * CONSTANTS_ONE_G;
 
 	// Calculate the specific total energy lower rate limits from the min throttle sink rate
-	_STE_rate_min = - _min_sink_rate * CONSTANTS_ONE_G;
+	const float rate_min = - _min_sink_rate * CONSTANTS_ONE_G;
+
+
+	const float lift = _auw * CONSTANTS_ONE_G;
+
+	// _Cd_i_specific: Vehicle specific induced drag coefficient, which equals to 1/2*S*rho*Cd_i
+	// Cd_i_specific = ... assuming planar wing with elliptical lift distribution
+	const float _Cd_i_specific = lift * lift / (0.5f * M_PI_F * CONSTANTS_AIR_DENSITY_SEA_LEVEL_15C * _wingspan * _wingspan);
+
+	// _Cd_o_specific: Vehicle specific parasitic drag coefficient, which equals to 1/2*A*rho*Cd_o
+	// Cd_o_specific: subtracting induced drag from total drag at a known airspeed to calculate parasitic drag
+	const float _Cd_o_specific = (-rate_min - _Cd_i_specific / _indicated_airspeed_trim) / (_indicated_airspeed_trim * _indicated_airspeed_trim * _indicated_airspeed_trim);
```

Comment:
@almaaro Isn't there a unit mismatch here? rate_min is an energy rate (or power) and has the unit m/s * m/s^2
_Cd_i_specific / _indicated_airspeed_trim according to your previous calculations has the unit kg*m^3 / s^4

Reply:
Definitely a point there. Here's what I came up with (bold for units that cancel out):
[_Cd_i_specific] = kg * m/s² * kg * m/s² * 1/(kg/m³) * 1/m * 1/m = kg * **m**/s² * kg * **m**/s² * 1/(kg/m³) * 1/**m** * 1/**m** = **kg**/s² * kg/s² * 1/(**kg**/m³) = kg * m³/s⁴

that divided by [_indicated_airspeed_trim] (m/s) gives kg*m²/s³.

So there is the extra kilogram. Let's see.

The starting equation:
`_min_sink_rate * CONSTANTS_ONE_G * _auw = _Cd_i_specific / EAS + _Cd_o_specific * EAS^3`
since
`rate_min = _min_sink_rate * CONSTANTS_ONE_G`
the `_Cd_i_specific` and `_Cd_o_specific` should be divided by _auw. But since `_Cd_o_specific` is calculated inderectly from `rate_min` and `_Cd_i_specific`, there is nothing to be divided while calculating it. 

So it should be
`const float _Cd_i_specific = lift * lift / (0.5f * M_PI_F * CONSTANTS_AIR_DENSITY_SEA_LEVEL_15C * _wingspan * _wingspan * _auw);`

@RomanBapst do you think this is valid?

I'll also change the underscoring to match local variable conventions.